src/main/java/slimeknights/tconstruct/gadgets/item/slimesling/MagmaSlimeSlingItem.java

```diff
@@ -31,49 +34,42 @@ public void onPlayerStoppedUsing(ItemStack stack, World worldIn, LivingEntity en
     AxisAlignedBB bb = player.getBoundingBox().expand(look.x * range, look.y * range, look.z * range).expand(1, 1, 1);
-    double minX = Math.min(start.x, direction.x);
-    double maxX = Math.max(start.x, direction.x);
-    double minY = Math.min(start.y, direction.y);
-    double maxY = Math.max(start.y, direction.y);
-    double minZ = Math.min(start.z, direction.z);
-    double maxZ = Math.max(start.z, direction.z);
-    List<Entity> entitiesInArea = player.getEntityWorld().getEntitiesWithinAABBExcludingEntity(player, bb);
-    double dist = range;
-    Entity closestEntity = null;
-    for (Entity entity : entitiesInArea) {
-      if (entity.getBoundingBox().intersects(minX, minY, minZ, maxX, maxY, maxZ)) {
-        if (look.distanceTo(entity.getLookVec()) < dist) {
-          dist = look.distanceTo(entity.getLookVec());
-          closestEntity = entity;
-        }
-      }
-    }
-
-    // cancel if there's a block in the way
-    BlockRayTraceResult mop = rayTrace(worldIn, player, RayTraceContext.FluidMode.NONE);
-    if (mop.getType() == RayTraceResult.Type.BLOCK && dist * dist < mop.getPos().distanceSq(player.getPosX(), player.getPosY(), player.getPosZ(), true)) {
-      return;
-    }
 
-    if (closestEntity != null) {
-      if (closestEntity instanceof LivingEntity) {
-        ((LivingEntity) closestEntity).applyKnockback(f , -look.x, -look.z);
+    EntityRayTraceResult emop = ProjectileHelper.rayTraceEntities(worldIn, player, start, direction, bb, (e) -> e instanceof LivingEntity);
+    if (emop != null && emop.getEntity() != null) {
+      LivingEntity target = (LivingEntity) emop.getEntity();
+      double targetDist = start.squareDistanceTo(target.getEyePosition(1F));
 
-        // TODO: Test on other players (I wasn't able to figure out setting that up)
-        if (closestEntity instanceof ServerPlayerEntity) {
-          ServerPlayerEntity playerMP = (ServerPlayerEntity) closestEntity;
-          TinkerNetwork.getInstance().sendTo(new EntityMovementChangePacket(closestEntity), playerMP);
-        }
+      // cancel if there's a block in the way
+      BlockRayTraceResult mop = rayTrace(worldIn, player, RayTraceContext.FluidMode.NONE);
+      double blockDist = mop.getPos().distanceSq(start.x, start.y, start.z, true);
+      if (mop.getType() == RayTraceResult.Type.BLOCK && targetDist > blockDist) {
+        playMissSound(player);
+        return;
+      }
 
-        player.playSound(Sounds.SLIME_SLING.getSound(), 1f, 1f);
+      target.applyKnockback(f , -look.x, -look.z);
+      if (target instanceof ServerPlayerEntity) {
+        ServerPlayerEntity playerMP = (ServerPlayerEntity) target;
+        TinkerNetwork.getInstance().sendTo(new EntityMovementChangePacket(target), playerMP);
       }
+
+      player.playSound(Sounds.SLIME_SLING.getSound(), 1f, 1f);
+    } else {
+      playMissSound(player);
     }
   }
+
+  // Using function for consistency
+  public void playMissSound(PlayerEntity player) {
```

Comment:
Since its just for consistency, can probably be private. I doubt anyone will extend this, but I'd rather make too much private than too much public.